service/resource/configmap/current_test.go

```diff
@@ -0,0 +1,94 @@
+package configmap
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/pkg/api/v1"
+
+	"github.com/giantswarm/micrologger/microloggertest"
+)
+
+// TestGetCurrentState tests the GetCurrentState method.
+func TestGetCurrentState(t *testing.T) {
+	configMapNamespace := "monitoring"
+	configMapName := "prometheus-config"
+
+	tests := []struct {
+		setUp             func(kubernetes.Interface) error
+		expectedConfigMap *v1.ConfigMap
+	}{
+		// Test that when the configmap does not exist, GetCurrentState returns nil.
+		{
+			setUp: func(k8sClient kubernetes.Interface) error {
+				return nil
+			},
+			expectedConfigMap: nil,
+		},
+
+		// Test that when the configmap does exist, GetCurrentState returns the configmap.
+		{
+			setUp: func(k8sClient kubernetes.Interface) error {
+				_, err := k8sClient.CoreV1().ConfigMaps(configMapNamespace).Create(&v1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      configMapName,
+						Namespace: configMapNamespace,
+					},
+					Data: map[string]string{
+						"foo": "bar",
+					},
+				})
+
+				return err
+			},
+			expectedConfigMap: &v1.ConfigMap{
```

Comment:
I like the tests for the current state. I was wondering how to do it in other resources. Good to know how you did it here.